java_cert module: add 'trustcacerts' option for certificate import #37707

gyptazy opened the pull request:
#37707  added option to add trusted cert to keystore

##### SUMMARY
This is an implementation of feature request #37707 to add certificates as trusted cacert.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
java_cert module

##### ANSIBLE VERSION
```
ansible 2.4.2.0
```

florianpaulhoberg committed:
added option to add trusted cert to keystore

 added option to add trusted cert to keystore

florianpaulhoberg committed:
fixed trailing whitespace

 fixed trailing whitespace

haad commented:
shipit

haad commented:
thanks for your patch!

vandrade commented:
It looks good and seems like a basic feature, what is holding up this PR?

haad commented:
shipit

anaeh commented:
Please merge the feature.

florianpaulhoberg commented:
Any updates?

florianpaulhoberg commented:
@haad ping

haad commented:
We need some other maintainer to mark it as mergable. Right now only me marked it that way.

pholica committed:
os_image_facts: add ability to filter Openstack images based on properties

alvarow commented:
Please merge!

Qalthos committed:
Move instantiation of terminal plugin earlier (#61261)

* Move instantiation of terminal plugin earlier

* Move test for invalid network_os to where the failure now lives

goneri committed:
ansible-test: async deployment w/ vcenter provider (#59984)

ansible-test: async deployment w/ vcenter provider

So far, `ansible-test`'s vcenter provider was trigger by a blocking POST
call. Since the depoyment take a lot of time (> 8m), we decided to use
an async call instead and poll the API server until the deployment
is ready.

dress-socks committed:
Update _ios_l2_interface.py (#61401)

Add support for TwoGigabitEthernet switchports in IOS-XE

felixfontein committed:
vertica_facts: rename to vertica_info (#60916)

* Rename vertica_facts -> vertica_info

* Add changelog.

* Fix porting guide.

samdoran committed:
Revert "Disable Hetzner Cloud integration tests due to authentication issues (#61375)" (#61383)

This reverts commit 31b49334b99fbce799d5ba537bd29f69dc42c849.

pabelanger committed:
Remove unsed / duplicate imports for iosxr module_utils (#61385)

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

pabelanger committed:
Remove unused import in junos/utils/utils.py (#61384)

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

kbreit committed:
Meraki - Enable API call rate limiting for requests (#54827)

* Initial commit for rate limiting
- Detects if error code is 429
- Pauses for random time between .5 and 5 seconds before retrying
- If it fails 10 times, give up and tell user

* Redo structure of request() to support rate limiting

* Hold down timer is now a sliding scale
- 3 * number of retries
- Fails after the 30 second wait

* Whitespace fixes

* Redo implementation using decorators
- Errors aren't tested but code works for regular calls

* Unit tests work for error handling

* Add integration tests for successful retries

* Add condition for 502 errors and retry

* Move _error_report out of the class

* PEP8 fixes

* Add changelog entry

felixfontein committed:
Online: rename _facts -> _info (#61091)

* Rename online _facts -> _info

* Add changelog.

felixfontein committed:
Scaleway: rename _facts -> _info (#61095)

* Rename scaleway _facts -> _info

* Add changelog.

* Improve docs.

felixfontein committed:
smartos_image_facts: rename to smartos_image_info (#60915)

* Rename smartos_image_facts -> smartos_image_info

* Add changelog.

smile37773 committed:
Rename Azure modules:  facts -> info (#60987)

jborean93 committed:
PowerShell - Added coverage collector (#59009)

* Added coverage collection for PowerShell - ci_complete ci_coverage

* uncomment out coverage uploader call

* Generate XML for PowerShell coverage

* Use whitelist to exclude coverage run on non content plugins

* Remove uneeded ignore entry

* Try to reduce diff in cover.py

* Fix up coverage report package - ci_complete ci_coverage

ericsysmin committed:
added gcp_kms filters filter plugin (#56755)

* added gcp_kms filters filter plugin

* update filter header

* remove requests

acozine committed:
removes the word 'support' from appendix pages (#61409)

felixfontein committed:
OneView: rename _facts -> _info (#61208)

* Rename OneView _facts modules -> _info

* Adjust PR #.

* Forgot to update test names.

* Remove superfluous blank line.

* Some more things from review.

smile37773 committed:
Add the azure_rm_azurefirewall_info module (#58697)

* add the info module of azure firewall

* fix the format_item()

* add the test

* fix the indented line

* fix

* fix the test

* change the test file

yuwzho committed:
Add azure resource lock module (#55700)

* add locks

* rename the module

* add test

* add test

* address comments

* add quote

* can list child scope lock

* minor docs tweaks

* Add files via upload (#62)

* change '\r\n' to '\n' (#63)

* Small changes, just to trigger CI verify.

* trigger CI verify

* remove 's'

* Update according by comments

* change small for trigger CI check

sdodsley committed:
New module to manage volume groups on Pure Storage FlashArray (#58854)

sdodsley committed:
New module for SMTP configuration on Pure Storage FlashBlade (#59270)

sdodsley committed:
New module to set alert email addresses for Pure Storage FlashArray (#57864)

sdodsley committed:
New module to set SMTP config for Pure Storage FlashArray (#58899)

sdodsley committed:
Changes references to ansible_facts (FB_S3USER) (#61356)

sdodsley committed:
Changes references to ansible_facts (FA_RA) (#61355)

sdodsley committed:
Change references to ansible_facts (#61353)

acozine committed:
remove the word "support" from Windows pages, add Windows Server 2019 (#61404)

* remove support from Windows pages, add Windows Server 2019

* jborean feedback

* Removed CBT info

Removed the CBT note about what transports Ansible actually supports. We've worked with both NTLM and Kerberos for a while now.

sergioirium committed:
Added Notes (#61315)

##### SUMMARY
<!--- Your description here -->


##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

sdodsley committed:
New module for Pure Storage FlashBlade Remote Assist control (#59261)

sdodsley committed:
New module to manage SNMP managers in Pure Storage FlashArray (#59203)

sdodsley committed:
New module to allow two FlasArrays to connect ready for replication (#60293)

briantist committed:
Allow password to be null in Scheduled Task for gMSA (#60990)

* Allow password to be null in Scheduled Task for gMSA

* Remove test for removed password validation, linting fixes

jborean93 committed:
Change collection PS util import pattern (#61307)

* Change collection PS util import pattern

* Add changes for py2 compat

* fix up regex and doc errors

* fix up import analysis

* Sanity fix for 2.6 CI workers

* Get collection util path for coverage collection

ShachafGoldstein committed:
win modules - standarize regex  and regxp as alias (#59037)

* win modules - standarize regex  and regxp as alias

* Changes to reflect desired state

* Update parameter names of funtions

* Update main.yml

* Update win_lineinfile.py

* Update win_wait_for.py

Ruben-Bosch committed:
win_updates: Add flag to only download updates without installing them (#58631)

* win_updates: Add flag to only download updates without installing them

* Fix test

* Fixes ansible-test (pep8)

* Fix integration test

* Fix actual fix.

mattclay committed:
Support relative imports in AnsiballZ. (#61196)

ericsysmin committed:
added gcp_storage_file lookup plugin (#56754)

* added gcp_storage_file lookup plugin

* updated to remove shebang and clean header

* clean header

sushma-alethea committed:
Branch icx l3 interface new (#61363)

* new module

* icx_cliconf

* icx test units module

* icx test units module

* icx units module

* icx units module

* icx banner unit test

* added notes

* new changes

* New branch to prevent complex rebase

* Rebase had problems

sushma-alethea committed:
icx: new module icx_interface (#60037)

* new module

* new module

* new module

* new terminal

* new terminal

* new terminal

* new terminal

* new terminal

* new terminal

* new cliconf

* new cliconf

* cliconf

* cliconf

* icx cliconf

* icx cliconf

* icx_cliconf

* icx_cliconf

* icx test units module

* icx test units module

* icx units module

* icx units module

* icx banner unit test

* icx banner unit test

* icx banner unit test

* PR changes resolved

* PR changes resolved

* changes resolved

* changes resolved

* Changes Resolved

* Changes Resolved

* check_running_config changes resolved

* check_running_config changes resolved

* check_running_config changes resolved

* added notes

* added notes

* new changes

* PR comments resolved

* new module icx_interface

* new module icx_interface

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* new fix

* new fix

* new fix

* new fix

* new fix

* new fix

* new fix

* new fix

* new fix

* new fix

* new fix

* new changes

* new changes

* new changes

* new changes

* new changes

* new changes

* notes updated

* Update icx.py

* Whitespace error

ganeshrn committed:
Add support for network_cli connection retry (#61103)

* Add support for network_cli connection retry

*  Add network_cli connection configuration option
   to allow retrying the connection initialization
   with remote host.

* Add docs

* Fix test failures

* Fix review comments

Commit:
Adding snapshot module (#61418)

* adding snapshot module

* adding snapshott module

* updates

* updates

* positional args

mattclay committed:
Overhaul ansible-test test path handling. (#61416)

* Remove .keep files from test/results/ dirs.

* Remove classification of test/results/ dir.

* Add results_relative to data context.

* Use variables in delegation paths.

* Standardize file writing and results paths.

* Fix issues reported by PyCharm.

* Clean up invocation of coverage command.

It now runs through the injector.

* Hack to allow intercept_command in cover.py.

* Simplify git ignore for test results.

* Use test result tmp dir instead of cache dir.

* Remove old .pytest_cache reference.

* Fix unit test docker delegation.

* Show HTML report link.

* Clean up more results references.

* Move import sanity test output to .tmp dir.

* Exclude test results dir from coverage.

* Fix import sanity test lib paths.

* Fix hard-coded import test paths.

* Fix most hard-coded integration test paths.

* Fix PyCharm warnings.

* Fix import placement.

* Fix integration test dir path.

* Fix Shippable scripts.

* Fix Shippable matrix check.

* Overhaul key pair management.

migumun committed:
FortiOS modules for 2.9 - 11 (#61387)

Andersson007 committed:
new module postgresql_publication (#59439)

* new module postgresql_publication: initial commit

* new module postgresql_publication: add seealso section

* new module postgresql_publication: fix RETURN section, fix typo

* new module postgresql_publication: add __exec_sql method

* new module postgresql_publication: fix typos

* new module postgresql_publication: fix typos

* new module postgresql_publication: fix typos

smile37773 committed:
Rename Azure modules: "facts"->"info"  (#61417)

* rename

* add symlink

* add the log

* fix the ignore file

* fix

* fix

jborean93 committed:
Win setup processor (#61425)

* Fix incorrect processor information

Fixes Issue #45869 
Windows Processor Information Incorrect from Setup

* set ansible_processor as list of all cores and threads (mirroring POSIX facts)

* change to $win32_cs.NumberOfLogicalProcessors to calculate $cpu_list

* Simplify vcpus count source

Co-authored-by: Jesse Kaufman <jesse.kaufman@gmail.com>

sushma-alethea committed:
icx: new module icx_lldp (#61310)

* new module

* icx_cliconf

* icx test units module

* icx units module

* added notes

* new changes

* new module

* new fix

* new changes

* new fixes

* new changes

* Rebase

jillr committed:
Deprecate alias 'thirsty' from all usages (#61245)

* Deprecate alias 'thirsty' from all usages

Fixes: #61236

* Now with added version quoting

* Handle deprecated aliases in mod_utils

* Make alias deprecation a subkey of the canonical arg, and not a separate argument

anas-shami committed:
Feature/onyx wjh upstream (#61269)

* support wjh module config with testing file

* Fix version, fix enable/disable all groups

* refactor get current group status

* fix indentation and change command_generate function, add test cases and check them

* Update onyx_wjh.py

fix description

* Update onyx_wjh.py

remove extra spaces, add type to choices after run validate-modules

* fix format string - pytest use 2.6.6 format function

* Remove unused imports

* chenge onyx_wjh file permessions

* Fix version, fix enable/disable all groups

* fix indentation and change command_generate function, add test cases and check them

* support wjh module config with testing file

* Fix version, fix enable/disable all groups

* refactor get current group status

* fix indentation and change command_generate function, add test cases and check them

* Update onyx_wjh.py

fix description

* Update onyx_wjh.py

remove extra spaces, add type to choices after run validate-modules

* fix format string - pytest use 2.6.6 format function

* fix format string - pytest use 2.6.6 format function

* chenge onyx_wjh file permessions

ujwalkomarla committed:
Fix exos_facts httpapi (#61198)

fortinetps committed:
Files from PR#60621, continue working on fortios_facts modules (#61405)

* Files from PR#60621, continue working on fortios_facts modules

* PR#61405 unit test module test_fortios_facts.py

* sanity fixed in test_fortios_facts unit test module

ruturajvy committed:
Added integration tests for the following modules: (#55523)

- exos_facts
- exos_command
- exos_config

carchi8py committed:
Auto support logs were not being created (#60901)

* Auto support logs were not being created

* fixes

carchi8py committed:
New options to Interface  (#60499)

* update to interface

* force ansibot to run again

* fixes

samccann committed:
Update Ansible sphinx theme to rtd theme version 0.4.3 (#61189)

* update theme.conf, header, footer, etc.
* clean up layout.html
* move large Ansible changes out of theme.css
* maxify theme to help merge
* remove redundancies, fix top nav bar
* add Ansible color to header and left-nav menu

Qalthos committed:
New module eos_lldp_interfaces (#61341)

* Prepare existing files for lldp_interfaces

* Initial commit of scaffolding

* Implement facts

* Implement config

* Add tests

* Appease Shippable

Yannis100 committed:
Typo in example for ios_l2_interface - allow_vlans (#61366)

swartzn committed:
Add tray/drawer loss protection preference and integration tests to netapp_e_storagepool module. (#60151)

mattclay committed:
Add a --venv option to ansible-test. (#61422)

* Add --venv delegation to ansible-test.

* Update import sanity test venv creation.

* Fix import test when using --venv on Python 2.x.

* Improve virtualenv setup overhead.

* Hide pip noise for import sanity test.

* Raise verbosity on venv info messages.

* Get rid of base branch noise for collections.

* Add missing --requirements check.

chkp-orso committed:
Checkpoint Network module (#61309)

* network module

* delete cp_network, cp_network_facts, and change test_cp_network to match the new network module

* Update ignore.txt

* adding state: present

* update example

* list to dict

* Update test_cp_mgmt_network.py

* remove white-space

chkp-orso committed:
publish module (#61312)

chkp-orso committed:
address_range module (#61273)

* adress_range module

* adding '-'

*  adding state: present

* update example

* list to dict

chkp-orso committed:
assign_global_assignment module (#61290)

* assign_global_assignment module

* add short description

* add '-'

geragio committed:
filename attribute (#61426)

Adds missing return key/value 'filename' to the win_find module documentation

Spencer-Doak committed:
Correct grammar of contribute-message in footer. (#61414)

Adds a comma to the "you can edit this documentation" message in the footer on every documentation page.

mattclay committed:
Attempt quick fix for CI failures.

mattclay committed:
Fix CI test result placement.

Tomorrow9 committed:
VMware: vmware_guest_screenshot module fix get_parent_datacenter issue (#60626)

Fix get_parent_datacenter not found issue, and change "local_path" parameter to "path" type.

closes: #60565

Qalthos committed:
New module eos_lacp_interfaces (#61349)

* Prepare for lacp_interfaces.

* Import module from builder

* Implement facts

* Implement config

* Add tests

samdoran committed:
Add listen_port_facts module (#61463)

Co-authored-by: Nathan Davison <ndavison85@gmail.com>

modular-magician committed:
Bug fixes for GCP modules (#61359)

Qalthos committed:
Fixes #61244 (#61432)

ujwalkomarla committed:
Resource Module for exos_lldp_global (#60988)

* Model->Role->Module.
Yet to import modules/exos_facts.py

* exos_facts refactor

* future-import and metaclass

* Fix unit tests

* Fix openconfig-lldp word and send_request API from Httpapi class

idempotent, json.dumps in base class, send_requests from base exos class

* action file for exos_lldp_global

* Add intergration tests for exos_lldp_global

* Test data field of the request

migumun committed:
Minor bug fixes - 1 (#61454)

migumun committed:
Minor bug fixes - 6 (#61464)

migumun committed:
Minor bug fixes - 1 (#61456)

yarix committed:
improve list paremeter description. (#60948)

* improve list option description.

felixfontein committed:
oVirt: rename _facts -> _info (#61299)

* Rename oVirt _facts -> _info

* Adjust PR #.

* Forgot update BOTMETA / ignore.txt for doc fragment.

* Compatibility for 3rd-party modules using module_utils or info doc fragment.

* Update note.

* Fix/improve docs.

* Update lib/ansible/modules/cloud/ovirt/ovirt_event_info.py

Co-Authored-By: Sloane Hertel <shertel@redhat.com>

* Fix variable.

* Fix return value documentation for ovirt_external_provider_info.

migumun committed:
Minor bug fixes - 5 (#61462)

migumun committed:
Minor bug fixes - 3 (#61461)

mattclay committed:
Revert "maven_artifact.py - add support for version ranges by using spec (#54309)"

This reverts commit 145a6a8756e0c02f565fe789c7ab2867d677b6f9.

alikins committed:
Support galaxy v3/autohub API in ansible-galaxy (#60982)

* Add galaxy collections API v3 support

Issue: ansible/galaxy-dev#60

- Determine if server supports v3

Use 'available_versions' from `GET /api`
to determine if 'v3' api is available on
the server.

- Support v3 pagination style

ie, 'limit/offset style', with the paginated
responses based on https://jsonapi.org/format/#fetching-pagination

v2 galaxy uses pagination that is more or less
'django rest framework style' or 'page/page_size style',
based on the default drf pagination described
at https://www.django-rest-framework.org/api-guide/pagination/#pagenumberpagination

- Support galaxy v3 style error response

The error objects returned by the galaxy v3 api are based
on the JSONAPI response/errors format
(https://jsonapi.org/format/#errors).

This handles that style response. At least for publish_collection
for now. Needs extracting/generalizing.

Handle HTTPError in CollectionRequirement.from_name()
with _handle_http_error(). It will raise AnsibleError
based on the json in an error response.

- Update unit tests

update test/unit/galaxy/test_collection*
to paramaterize calls to test against
mocked v2 and v3 servers apis.

Update artifacts_versions_json() to tale an
api version paramater.

Add error_json() for generating v3/v3 style error
responses.

So now, the urls generated and the pagination schema
of the response will use the v3 version if
the passed in GalaxyAPI 'galaxy_api' instance
has 'v3' in it's available_api_versions

* Move checking of server avail versions to collections.py

collections.py needs to know the server api versions
supported before it makes collection related calls,
so the 'lazy' server version check in api.GalaxyAPI
is never called and isn't set, so 'v3' servers weren't
found.

Update unit tests to mock the return value of the
request instead of GalaxyAPI itself.

samdoran committed:
validate-modules - Use error codes instead of numbers (#60711)

swartzn committed:
Add netapp_e_drive_firmware (#59528)

* Add na_santricity_drive_firmware module

Manage NetApp E-Series drive firmware downloads
Includes unit and integration tests

* Rename na_santricity_drive_firmware to netapp_e_drive_firmware

swartzn committed:
Add netapp_e_firmware module (#59529)

* Add na_santricity_firmware module.

Manages NetApp E-Series firmware upgrades.
Includes unit and integration tests.

* Add legacy support to na_santricity_firmware module.

* Rename na_santricity_firmware to netapp_e_firmware

* Improved netapp_e_firmware example documentation.

samdoran committed:
Show doc URL for sanity test failures

ctrufan committed:
Addition of ECS_Certificate Module (#60883)

* Addition of ecs_certificate module.

* Documentation and code fixes

* Updates per code review

* Doc fixes, rename of chain_path to full_chain_path, add regex for cert_Expiry check

* Fixes to pep8 check to make regexp string 'raw'.

* Mistakes with find/replace of caseing.

* Added integration tests and some doc cleanup

* Some additional assertions and test typo cleanup

* Update lib/ansible/modules/crypto/entrust/ecs_certificate.py

Co-Authored-By: Felix Fontein <felix@fontein.de>

* Responses to code review comments

* Remove fake passwords from aliases file.

zeten30 committed:
New storage/netapp module - na_ontap_kerberos_realm (#59923)

* New module na_ontap_kerberos_realm

* Copyright updated

* Unit test for modules + fixed required_if parameters issue

* reverting state=modify

* fixing test

* Fixing imports section

mattclay committed:
Change location of collections tests.

The directories used for collections tests are changing as follows:

`test/` -> `tests/`
`test/results/` -> `tests/output/`

This is a breaking change for collections tests executed by ansible-test.
All collections will need to be updated to use the new directory.

The `tests/output/` directory should be added to the `.gitignore` or equivalent in each collection.

This change is being made before the first pre-release of Ansible 2.9 since ansible-test has not yet been shipped.
Using the `tests/` directory matches the other collections directories `plugins/` and `roles/`.

This resolves https://github.com/ansible/ansible/issues/60218

mattclay committed:
More ansible-test path handling updates. (#61484)

* Move test results vars into layout.

* Add test_path to layout.

* Add sanity_path to layout.

* Clean up layout.

* Update change classification.

* Improve classification layout.

* Relocate common classification code.

* Use is_subdir for relocated code.

* Relocate ansible test/units/compat/ hack.

* Fix variable shadowing.

* Remove unused code.

* Fix ordering of path tests.

* Clean up ansible classification.

* Fix hard-coded plugin paths in classification.

* Relocate more common classification.

* Fix logic.

* Fix pylint issue.

* Add missing licenses classification.

mattclay committed:
Improve ansible-test --venv error handling.

This allows it to fall back to virtualenv when venv is broken.

robertchung committed:
Fix TypeError in ec2_group.py for Python3 when sorting dictionary list (#59844)

* Fix TypeError in ec2_group.py for Python3 when sorting dictionary list

* Using json.loads() and dumps() to replace sorting

* Bug fixes for ec2_group.py

* Dictionaries cannot be compared/sorted in Python3

* Diff will occur when the IpPermissions have the same IpRanges but have different ordering

* 'before' will be sorted by 'Type' with high priority than 'IP', but 'boto3.describe_security_groups()' function cannot get 'Type' from Amazon

* Add some basic diff mode testing to exercise the rule-sorting code

nitzmahone committed:
default collection support (#61415)

* default collection support

* playbooks run from inside a registered collection will set that collection as the first item in the search order (as will all non-collection roles)
* this allows easy migration of runme.sh style playbook/role integration tests to collections without the playbooks/roles needing to know the name of their enclosing collection

* ignore bogus sanity error

* filed #61460

* fixed task unit test failure

* don't append an empty collections list to the ds

* ignore leftover local_action in mod_args ds action parsing

* fix async_extra_data test to not require ssh and bogus locale

* disable default collection test under Windows

* ensure collection location FS code is always bytes

* add changelog

jimi-c committed:
T woerner max concurrent (#60702)

* play, block, task: New attribute forks

With this it is possible to limit the number of concurrent task runs.
forks can now be used in play, block and task. If forks is set in different
levels in the chain, then the smallest value will be used for the task.

The attribute has been added to the Base class as a list to easily provide
all the values that have been set in the different levels of the chain.

A warning has been added because of the conflict with run_once. forks will
be ignored in this case.

The forks limitation in StrategyBase._queue_task is not used for the free
strategy.

Signed-off-by: Thomas Woerner <twoerner@redhat.com>

* Handle forks in free strategy

The forks attribute for the free strategy is handled in run in the free
StrategyModule. This is dony by counting the amount of tasks where the uuid
is the same as the current task, that should be queued next. If this amount
is bigger or equal to the forks attribute from the chain (task, block,
play), then it will be skipped to the next host. Like it is also done with
blocked_hosts.

Signed-off-by: Thomas Woerner <twoerner@redhat.com>

* Test cases for forks with linear and free strategy

With ansible_python_interpreter defined in inventory file using
ansible_playbook_python.

Signed-off-by: Thomas Woerner <twoerner@redhat.com>

* Changing forks keyword to throttle and adding some more docs

felixfontein committed:
Reordering entries. (#61491)

mattclay committed:
Fix paths in sanity tests.

carchi8py committed:
new module: Cloud Volume for AWS FileSystems (#61343)

* new module"

* fixes

* updates

carchi8py committed:
new module netapp csv pools for aws (#61340)

* new module cvs pools

* fixes

* updates

* fixes

* fixes

* fixes

migumun committed:
Minor bug fixes - 3 (#61458)

chkp-orso committed:
threat_rule module (#61280)

* threat_rule module

* remove :

* insert to list

* remove redundant indentation

* Revert "insert to list"

This reverts commit de08dfdf3221de9ae82c4014de2436f77c46401e.

* enable longer lines, remove present, add '-'

* state: present

* update examples

* list to dict

* remove rule_number

chkp-orso committed:
access_rule module (#61281)

* access_rule module

* remove :

* fix to pass tests

* don't start line with quote (")

* remove redundant indentation

* return the origin description

* don't start line with quote (")

* enable longer lines, add '-'

*  adding state: present

* update examples

* dict to list

* list to dict

* Update cp_mgmt_access_rule.py

* remove rule_number

chkp-orso committed:
application_site module (#61287)

* application_site module

* add '-'

* update examples

* list of objects

* fix example

chkp-orso committed:
group module (#61274)

* group module

* add '-'

* update examples

* fix

chkp-orso committed:
service_tcp module (#61275)

* service_tcp module

* add '-'

* update examples

* fix

chkp-orso committed:
service_udp module (#61276)

* service_udp module

* add '-'

* update examples

* fix

chkp-orso committed:
host module (#61277)

* host module

* add '-'

* update examples

* fix

chkp-orso committed:
threat_indicator module (#61278)

* threat_indicator module

* remove _

* add '-'

* update examples

* fix

chkp-orso committed:
access_role module (#61282)

* access_role module

* add '-'

* update examples

* list to dict

zeten30 committed:
Improving na_ontap_command module (#60471)

* Improving na_ontap_command

- new option 'return_dict': true|false
- module returns parseable dictonary data instead of raw XML output when return_dict: true

* Fix module documentation typo

* Fixing pull request comments

* Python lint

* smaller try blocks

chkp-orso committed:
threat_protection_override module (#61481)

chkp-orso committed:
application_site_category module (#61288)

* application_site_category module

* add '-'

* update examples

* list to dict

chkp-orso committed:
administrator module (#61286)

* administrator module

* fix examples that the list param will apear with '-' instead of '_'

* add '-'

* update examples

* list to dict

zeten30 committed:
New module - na_ontap_ldap (#59718)

* New module - na_ontap_ldap

* Fixing module documentation - ansible-test sanity

* Fulfil merge request changes from comments

- 'client_config' option renamed to 'name'
- docstring updated
- imports placed according to Ansible module best practices

* Fixed docstring typo + netapp_utils.ems_log_event added

chkp-orso committed:
access_layer module (#61373)

* access_layer module

* list to dict

chrisvanheuveln committed:
nxos_bfd_interfaces: new module (#61407)

* bfd_interfaces: initial commit

* fix state methods, add more UT

* Add integration tests, support for N6K

* minor comment cleanups

* lint 1

* lint 2

* lint 3

* lint 4

* lint 5

* retrigger shippable

* retrigger shippable

* PR review updates: /bfd_echo/echo/, updated tests

* fix overridden logic and integration test

zeten30 committed:
New module - na_ontap_ldap_client (#59713)

* New module - na_ontap_ldap_client

* Fixing module example usage

* Fixing module documentation - ansible-test sanity

* Fulfile merge request comments

- required_if used for options
- 'client_config' option renamed to 'name'
- docstring updated
- imports placed according to Ansible module best practices

* create an ems log event for users with auto support turned on

* Fix required_if state=present options

chkp-orso committed:
time module (#61477)

chkp-orso committed:
put_file module (#61479)

chkp-amiads committed:
verify_policy module (#61455)

danielmellado committed:
Limit config fetch on junos_l3_interfaces (#61219)

This commit adds a filter in junos_l3_interfaces so it limits the amount
of configuration fetched from the remote device.

chkp-amiads committed:
service_sctp module (#61386)

* service_sctp module

* list to dict

* fix example

chkp-amiads committed:
simple_gateway module (#61388)

* simple_gateway module

* list to dict

* fix indentation issue

fails tests

chkp-amiads committed:
package module (#61439)

carchi8py committed:
new module: Cloud Volumes for AWS, active Directory (#61342)

* new module

* fixes

* update

chkp-amiads committed:
service_icmp module (#61451)

* service_icmp module

* fix example

chkp-amiads committed:
service_other module (#61450)

* service_other module

* fix test failure

carchi8py committed:
New module: cvs snapshots for aws (#61023)

* New module: cvs snapshots for aws

* fixeS

chkp-amiads committed:
vpn_community_meshed module (#61397)

* vpn_community_meshed module

* list to dict

chkp-amiads committed:
vpn_community_star module (#61398)

* vpn_community_star module

* list to dict

chkp-amiads committed:
multicast_address_range module (#61438)

chkp-amiads committed:
threat_profile module (#61391)

* threat_profile module

* list to dict

* fix example

samdoran committed:
Fix race condition in test_sunos_get_uptime_facts

Mock time.time() to return consistent time value.

chkp-orso committed:
threat_exception module (#61470)

* threat_exception module

* fix examples

chkp-amiads committed:
install_policy module (#61446)

chkp-amiads committed:
run_script module (#61444)

chkp-amiads committed:
global_assignment module (#61449)

Commit:
SIG Additional fix (#61502)

* adding snapshot param

* fix problem with leftover empty dict

* leftovers

* fix leftovers

chkp-amiads committed:
dns_domain module (#61436)

* dns_domain module

* fix example

chkp-amiads committed:
wildcard module (#61399)

* wildcard module

* list to dict

* fix example

chkp-amiads committed:
service_dce_rpc module (#61465)

* service_dce_rpc module

* fix example

chkp-amiads committed:
service_rpc module (#61468)

* service_rpc module

* fix example

chkp-amiads committed:
service_icmp6 module (#61453)

* service_icmp6 module

* fix example

chkp-amiads committed:
security_zone module (#61447)

* security_zone module

* fix example

chkp-amiads committed:
exception_group module (#61474)

* exception_group module

* Update cp_mgmt_exception_group.py

ganeshrn committed:
Fix in get api for eric_eccli cliconf plugin (#61507)

Fixes https://github.com/ansible/ansible/issues/60674

*  Fix get() API signature for eric_eccli cliconf plugin
   to work with cli_command module.

mattclay committed:
Simplify ansible-test target processing. (#61506)

* Clean up layout paths for integration tests.

* Remove "special" integration test target type.

* Remove unnecessary role detection logic.

* Remove support for non-sh runme scripts.

* Simplify reading of aliases.

samdoran committed:
Add additional group for sanity tests

Put validate-modules in its own group

Commit:
Rename dtl and db _facts --> _info (#61496)

* rename modules

* created links

* fixed sanity

* removed unnecessary ignore rules

* added depreciation warning

* update ignore.txt to match

* removed unnecessary ignore rules

* update ignore.txt to match

* make ignore.txt same as in devel

* cr

* additional sanity

chkp-amiads committed:
service_group module (#61448)

* service_group module

* fix example (add)

* fix set example

chkp-amiads committed:
threat_layer module (#61389)

* threat_layer module

* list to dict

* fix example

* remove set example

ndclt committed:
keycloak: improve testability of authentification (#57611)

mattclay committed:
Move the ansible-test output_dir.

This directory is currently a fixed location to make troubleshooting easier.
It is cleared before each test target runs, but is preserved when a test target finishes.
This allows the contents to be inspected when a test fails.

The previous location was `~/ansible_testing/`.
The new location is within the content root:

- `test/results/.tmp/output_dir` for Ansible
- `tests/output/.tmp/output_dir` for Ansible Collections

Moving the directory reduces the number of places on the filesystem where tests create output.
It also enables the results to be returned from delegated systems.

chkp-amiads committed:
dynamic_object module (#61437)

* dynamic_object module

* fix example

* remove set example

chkp-amiads committed:
tag module (#61393)

* tag module

* list to dict

* removed set example

* fix name in example

smile37773 committed:
Rename Azure modules: facts -> info (#61505)

* rename

* add symlink

* rename

* rename

* rename

* rename

* rename

* rename

* rename

* rename

* rename

* rename

* rename

* rename

nitzmahone committed:
Collection role relative deps (#61517)

* default collection support

* playbooks run from inside a registered collection will set that collection as the first item in the search order (as will all non-collection roles)
* this allows easy migration of runme.sh style playbook/role integration tests to collections without the playbooks/roles needing to know the name of their enclosing collection

* disable default collection test under Windows

* enable collection search for role dependencies

* unqualified role deps in collection-hosted roles will first search the containing collection
* if the calling role has specified a collections search list in metadata, it will be appended to the search order for unqualified role deps

* disable cycle detection unit test

* failing on 3.7+, needs proper cycle detection
* see #61527

resmo committed:
vultr: add changelog and update porting guide (#61520)

chkp-amiads committed:
run_ips_update module (#61440)

mattclay committed:
Update collections integration targets path. (#61523)

* Update collections integration targets path.

* Fix integration path handling.

* Add collections test target fallback.

Also add warnings and errors for common path mistakes.

* Improve role target detection.

Andersson007 committed:
postgresql modules: add seealso section (#61526)

* postgresql modules: add seealso section

* postgresql modules: add seealso section

Andersson007 committed:
jboss: add seealso (#61532)

chkp-orso committed:
application_site_group (#61289)

* application_site_group

* add '-'

* update examples

* list to dict

* set example

chkp-amiads committed:
discard module (#61452)

* discard module

* fix test failure

chkp-amiads committed:
session module (#61471)

chkp-amiads committed:
mds module (#61472)

* mds module

* Update lib/ansible/modules/network/checkpoint/cp_mgmt_mds_facts.py

improve description

Co-Authored-By: Sumit Jaiswal <sjaiswal@redhat.com>

s-hertel committed:
add porting guide for bare vars in conditionals (#60943)

samdoran committed:
Don't bother Ricky (#61377)

Commit:
Rename remaining facts -> info (#61516)

* rename remaining stuff

* links

* fixed link

* sanity fixes

* too many lines

* fixed link

* fixed link again

* delete mistake

* fixed ignore

* fixed ltl

* fixed trailing whitespace

justjais committed:
Resource module for iosxr_l3_interfaces (#60464)

* iosxr_l3_interfaces module

chkp-orso committed:
group_with_exclusion module (#61480)

Qalthos committed:
Defer passlib dependency from plugins/filter/network.py (#59304)

* Defer passlib dependency from plugins/filter/network.py to what utils/encrypt.py provides.

* Revert test changes now that passlib is optional

chkp-orso committed:
fix suboptions of list to replacce _ with - (#61542)

* fix suboptions of list to replacce _ with -

* Update checkpoint.py

sivel committed:
Add porting guide entry for gathering facts tag change (#61180)

* Add porting guide entry for gathering facts tag change

* Use code-block

Yannis100 committed:
Add 'min' to gather_subset ios_facts documentation (#61536)

dpassante committed:
add tag functionality to cs_network (#61530)

* add tag functionality to cs_network

* fix version_added

resmo committed:
cloudscale_volume: refactor to implement tags support (#61531)

felixfontein committed:
CloudStack modules: rename _facts -> _info (#61090)

* Rename cloudstack _facts -> _info

* Add changelog.

* Fix errors due to overloaded get_result having different parameter names.

* Fix examples.

* Change debug tasks.

* Remove unneeded code.

* Change from rename -> deprecate+new module.

* Make cs_zone_info return a list.

* Make cs_instance_info return a list.

* Fix return value docs.

* Fix typo.

* Improve tests.

* Fix cs_zone_info.

* Linting.

* Fix alias/option switch.

* Fix version numbers.

* Extend docs.

ganeshrn committed:
Fix doc for network_cli connection retry timer (#61509)

ujwalkomarla committed:
Update BOTMETA to be included as maintainer for EXOS modules. (#61485)

Tomorrow9 committed:
VMware: add support for multiple CD-ROMs (#51120)

Add support for create or re-configure VM with multiple CD-ROMs attaching to IDE controller now, will implement SATA controller support later.
parameters can be set as below:

    cdrom:
    - controller_type: ide
      controller_number: 0
      unit_number: 0
      type: client

mkrizek committed:
BOTMETA: fix team_extreme name use (#61557)

carchi8py committed:
New Module: NetApp Account (#59939)

* netapp account module for azure

* netapp account module for azure

* fixes

* fixes

* add requirements

* add requirements

* fix issues

* add units

* add units

* updates

* updates

* fixes

* fixes

* fies

* fixes

abadger committed:
Revert "New Module: NetApp Account (#59939)"

This reverts commit 6a8dcb8fcbc0e79484067dc8c5b5dda173e73ce2.

Reverting this as it breaks CI for azure (more permissions would be
needed for our account)

pabelanger committed:
Properly load module_prefix for collections in task_executor.py (#60420)

* Properly load module_prefix for collections

Now that we are using collections for tasks, we need to properly split
the name and load the prefix properly.

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

* Add unit tests for task_executor

This commit adds missing unit tests for action handler in test_task_executor.

Signed-off-by: Daniel Mellado <dmellado@redhat.com>

ganeshrn committed:
Rename junos_lldp_global enable option to enabled (#61551)

*  Change `enable` option to `enabled` option to be in
   sync with other platforms

goneri committed:
ansible-test: vcenter+Worldstream: hide password

Hide the temporary password when `ansible-test` is called with the
`--redact` parameter.

sdodsley committed:
Fix crashing bug in purefa_info module and add some new info (#61494)

mattclay committed:
Fix ansible-test handling of submodules.

Resolves https://github.com/ansible/ansible/issues/61550

mikewiebe committed:
Rewrite nxos_file_copy as an action plugin (#60643)

* Initial nxos_file_copy action plugin work
* Remove code from nxos_file_copy module
* Add file_push and file_pull support
* Additional refactoring and shipable updates
* Simplify outcomes and update doc header
* Add more error data information for easier debugging
* Reorder outcomes and add additional tests
* Capture more data for permission denied outcome

modular-magician committed:
Bug fixes for GCP modules (#61512)

mattclay committed:
Use relative submodule status in ansible-test.

The `git submodule status` command is relative to the current git repository by default.
When running from a repository subdirectory paths can be returned above the current directory.
Specifying the current directory with `git submodule status` avoids listing submodules above that directory.

This will fix issues when testing a collection that is rooted below the repository root when that repository uses submodules.

jborean93 committed:
Refactor galaxy collection API for v3 support (#61510)

* Refactor galaxy collection API for v3 support

* Added unit tests for GalaxyAPI and starting to fix other failures

* finalise tests

* more unit test fixes

jborean93 committed:
Skip certain tests on 2016 (#61585)

sdodsley committed:
Add IOPs bandwidth support to volumes (#61577)

jborean93 committed:
Azure fix _info/_facts return values for some modules (#61584)

* Azure fix _info/_facts return values for some modules

* Further test fixes

* securitygroup fixes after the move to _info module

mattclay committed:
Fix ansible-test coverage path handling. (#61528)

* Fix ansible-test coverage path handling.
* Split CI unit tests into two groups.

s-hertel committed:
Run ec2_vol_facts integration tests in CI (#48755)

Filter owner aliases by 'amazon' since amazon marketplace may not be activated

abadger committed:
[WIP] Cleanups and version bumping for 2.10 (#61572)

* Cleanups and version bumping for 2.10

* Fix changelog url now that stable has been branched

* Fix the lenth of the porting guide title now that the version is two digits

acozine committed:
roadmap should reflect upstream release, not downstream (#61611)

tremble committed:
aws_codecommit: Fix integration tests and Add support for updating the description (#61263)

* Update DevOps AWS policy
- Fix typos in permission names
- While AWS claims you can use 'arn:aws:codecommit:*' it errors unless you use '*'

* aws_codecommit: (integration tests) Migrate to module_defaults

* aws_codecommit: (integration tests) Fix integration tests

* aws_codecommit: (integration tests) Add tests for updating the description

* aws_codecommit: Add support for updating the description and rename "comment" option to "description"

samccann committed:
fix width (#61619)

samdoran committed:
ansible-test - Continue if the git command returns an error (#61605)

* ansible-test - Contiune if the git command returns an error

* Just return stdout

* Use to_text() when displaying exception

* Add a message property to SubprocessError

samdoran committed:
Update Fedora 29 test image

resmo committed:
apt: use exponential backoff for apt update cache retries (#60527)

* make retries and max delay configurable

resmo committed:
apt: honor cache_update=false on auto-install deps (#56549)

* apt: honor cache_update=false on auto-install deps
* add porting guide

resmo committed:
apt_repository: implement update cache retrying (#57266)

* use exponential backoff
* make cache update retries and max delay in between configurable

goneri committed:
shippable: temporary bump the timeout for Azure jobs

justjais committed:
To fix iosxr_l3_interfaces module zuul tests (#61592)

* fix 61540

* fix utils

Qalthos committed:
Check module names in action plugin without collection attached (#60947)

* Check for eos_config in action plugin by module name, not entire fqmn

* Modify toher action plugins to find module name

* Restore missing `not`

* Cover netconf plugin as well

* Whoops

felixfontein committed:
Make sure postgresql tests are run for all postgres_* modules (#61647)

Andersson007 committed:
postgresql modules: explicit defaults for bool params (#61538)

leroyguillaume committed:
add option to define keysize at luks partition creation (#61522)

* add option to define keysize at luks partition creation

* Bump version_added.

* Extend tests.

pabelanger committed:
Remove unused import for iosxr l3_interfaces (#61563)

This was caught by our ansible network collections migration script.

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

mmazur committed:
kubevirt: also add wait_sleep (#61020)

carbolymer committed:
#59096 Fix for not working azure_rm_aks (#59097)

Fred-sun committed:
Set the default value for the managed_disk_type (#60749)

MaxBab committed:
os_server - add "tag" to instance nics (#61119)

A custom "tag" could be passed to the instance metadata with the nics.
Add support for the "tag" to the module.

gundalow committed:
Disable Automerge (#61674)

A few PRs have been automerged since devel became 2.10 even though they still had `version_added: 2.9`

Lets disable automerge till:
a) We find a proper fix (automerge should so `rebuild_merge`?)
b) We wait 7 days since devel became 1.10 to cause stale_ci

mkrizek committed:
perf: reduce getting handler variables (#59969)

* perf: reduce getting handler variables

ci_complete

* Do not re-create templar

fetzerch committed:
influxdb: Add support for 'path' parameter (#61173)

While it's possible to host InfluxDB on a different path prefix
(through a reverse proxy such as nginx or traefik), the Ansible
influxdb_* modules did't support that use case.

The patch adds a 'path' parameter, which is passed to the influxdb
Python client.

Example for connecting to InfluxDB running on http://localhost/influxdb

    - hosts: all
      tasks:
        - influxdb_database:
            database_name: test
            path: influxdb
            port: 80

vitalii-kostenko committed:
Ansible FTD Module improvements and tests update. (#60640)

* Ansible FTD Module improvements and tests update.

* Fix sanity tests

* Fix formatting

mulatinho committed:
luks_device.py: Allow manipulate LUKS containers with label or UUID (#61603)

* luks_device.py: Allow manipulate LUKS containers with label or UUID

- Allow create a LUKS2 container format with label support
- Allow manipulate (open, close, modify) an LUKS container based on
  both label (LUKS2 format) or UUID instead of using devices only.

Fixes: #58973
Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

* test_luks_device.py: organizing tests to support labels

- Add label on some tests and fix errors reported by Shippable

Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

* luks_device.py: adjusting versions and messages

- Modifying version_added from 2.9 to 2.10
- Fixing some messages
- Created a changelog fragment
- Moving blkid from scope

Fixes #58973
Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

Akasurde committed:
VMware: Fix get_tags_for_vm API in vmware_vm_info module (#61697)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

dgalpaj committed:
feat: support postgres 11 in azure_rm_postgresqlserver (#58727)

jotai73 committed:
Update homebrew_cask example docs (#61637)

Changed homebrew_cask module flag from "allow_external_apps" to "accept_external_apps" in example section

gundalow commented:
@florianpaulhoberg Hi, if you could rebase this I'll merge it.

Thank you everybody that's already reviewed this

gundalow commented on lib/ansible/modules/system/java_cert.py:
```diff
@@ -13,7 +13,7 @@
 ---
 module: java_cert
-version_added: '2.3'
+version_added: '2.4'
```
This shoudn't have changed

```suggestion
version_added: '2.3'
```

gundalow commented on lib/ansible/modules/system/java_cert.py:
```diff
@@ -32,6 +32,11 @@
+    description:
+      - Trust imported cert as CAcert
+    default: False
+    version_added: "2.6"
```
```suggestion
    version_added: '2.10'
```

gundalow commented on lib/ansible/modules/system/java_cert.py:
```diff
@@ -77,23 +82,29 @@
     executable: /usr/lib/jvm/jre7/bin/keytool
     state: absent
-
+- name: Import trusted CA from SSL certificate
```
Blank line between examples please
```suggestion

- name: Import trusted CA from SSL certificate
```

gundalow commented on lib/ansible/modules/system/java_cert.py:
```diff
@@ -77,23 +82,29 @@
+    keystore_create: yes
+    state: present
+    cert_alias: LE_RootCA
+    trust_cacert: True
```
```suggestion
    trust_cacert: True

```

gundalow commented on lib/ansible/modules/system/java_cert.py:
```diff
@@ -77,23 +82,29 @@
     keystore_pass: changeit
     keystore_create: yes
     state: present
-
```
Please keep the blank line

gundalow commented on lib/ansible/modules/system/java_cert.py:
```diff
@@ -110,13 +121,11 @@
   returned: success
   type: string
   sample: "Module require existing keystore at keystore_path '/tmp/test/cacerts'"
-
```
Please keep the blank line

gundalow commented on lib/ansible/modules/system/java_cert.py:
```diff
@@ -110,13 +121,11 @@
   returned: success
   type: int
   sample: "0"
-
```
Please keep the blank line

gundalow commented:
If you need help with rebasing please check have a look at this guide https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html

Qalthos committed:
Avoid infinite recursion on login (#61617)

willthames committed:
Remove willthames from AWS and Openshift notifications (#61670)

I no longer desire to be mentioned on most AWS changes.
If there are modules I'm on the authors list then I'll remain
notified.

Also remove me from openshift and kubectl plugins, I don't use
those.

Commit:
Fix network resource module wording and typographical errors (#61613)

liquidat committed:
Rename checkpoint to check_point due to legal reasons, fix #61145 (#61172)

* Fix #61145

* Fix type keyword

* Removed file fragment

t0mk committed:
Fix missing pass of IPXe params to API client call, fixes #32910 (#32912)

simon04 committed:
Fix typesetting of "GitLab" (#61702)

tahir-gvs committed:
Clarity on mutual exclusion of Username and Token (#61668)

##### SUMMARY
Although there is an example showing that username is not required with personal access token, it will be nice to more clearly state in options description. 
##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

it-praktyk committed:
Remove unnecessary information (#61657)

sestrella committed:
Update unit tests requirements/units.txt path (#61656)

resmo committed:
botmeta: unmaintain resmo from cloudstack (#61561)

Yannis100 committed:
net_user documentation typo configured_password (#61556)

password -> configured_password in the last example

amenonsen committed:
Remove duplicated word

We print "%s success", so there's too much success going on.

kbreit committed:
Docs - Add Meraki to platform option list (#61651)

it-praktyk committed:
Correct indentation of the code in the example (#61622)

ymartin59 committed:
Fix examples for bigip_file_copy (#61686)

wimnat committed:
Fix required_together statement - access_logs_s3_prefix is not required as per the documentation (#52017)

anshulbehl committed:
adding a collection local try section (#61626)

anryko committed:
datadog_monitor: Add process alert support (#60325)

theo-o committed:
grafana_plugin: update documentation (#60600)

samdoran committed:
Increase unit test timeout to account for long image download times (#61717)

cristidas committed:
Ecs taskdefinition secrets (#60112)

* adding specific examples for using environment and secrets in the container definition

gyptazy committed:
Merge branch 'patch-1' of github.com:florianpaulhoberg/ansible into patch-1

gyptazy committed:
* fixed merged conflicts
 * fixed syntax

florianpaulhoberg commented:
My fault while rebasing. @gundalow @haad please take a further look within the new PR in #61750. Thanks. After merging this task can be closed.

gundalow commented:
resolved_by_pr: 61750
Reviews welcome on https://github.com/ansible/ansible/pull/61750 so we can get that merged. Thanks in advance
